### PR TITLE
Tiny update of projection index

### LINKED ISSUE #368: 更新地图投影首页？
See https://docs.gmt-china.org/latest/proj/ and https://docs.generic-mapping-tools.org/dev/cookbook/map-projections.html.

- 中文手册将投影分为三类：笛卡尔**线性**投影、极坐标投影、地图投影
- GMT 官方手册使用 Miscellaneous projections 表示剩余地图投影，中文手册使用 Thematic
- 绘图时候将 **Other** 下包含了笛卡尔坐标的多种形式以及极坐标

感觉略微混乱，更新一下？

- [x] **笛卡尔线性投影** -->> **笛卡尔投影**
- [x] 图片中**Other**分类两列：**笛卡尔投影** 和 **极坐标投影**，与文字描述对应？
- [x] **Thematic** or **Miscellaneous**?

----

此外，我们是否增加投影汇总表格，参考 https://www.pygmt.org/latest/projections/index.html#projection-table 和 https://docs.generic-mapping-tools.org/dev/cookbook/map-projections.html 。当前只是一张图片展示分类。

### PR DESCRIPTION
**Pull request by core-man**
- Merge the two lists into a nested list
- Add some comments for each category

Fixes #368, Addresses #371.

### CODE REVIEW

File: source/proj/index.rst [hunk 1]
```diff
@@ -1,18 +1,22 @@
 地图投影
 ########
 
-GMT支持30余种不同的投影方式。这些投影方式可以分为三类：
+GMT 读取数据的实际坐标，并将其转换或者投影到图片的对应位置。GMT 支持 30 多种坐标变换或投影方式，
+这些投影方式可以分为三类，分类又细分为几个小类：
```
Comment (core-man) [inline source/proj/index.rst:5]: Refer to https://docs.generic-mapping-tools.org/dev/cookbook/coordinate-transformations.html:

> GMT programs read real-world coordinates and convert them to positions on a plot. This is achieved by selecting one of several coordinate transformations or projections. We distinguish between three sets of such conversions:
> - Cartesian coordinate transformations
> - Polar coordinate transformations
> - Map coordinate transformations

File: source/proj/index.rst [hunk 2]
```diff
@@ -1,18 +1,22 @@
...
 
-#. 笛卡尔线性投影
-#. 极坐标投影
-#. 地图投影
+#. 笛卡尔投影：细分为三类（位于下图的 OTHER 分类）
 
-其中，地图投影又可以细分为四大类：
+   - 笛卡尔线性投影
+   - 笛卡尔对数投影
+   - 笛卡尔指数投影
```
Comment (core-man) [inline source/proj/index.rst:11]: Refer to https://docs.generic-mapping-tools.org/dev/cookbook/coordinate-transformations.html#cartesian-coordinate-transformations:
> GMT Cartesian coordinate transformations come in three flavors:
> - Linear coordinate transformation
> - Logarithmic coordinate transformation
> - Power (exponential) coordinate transformation

File: source/proj/index.rst [hunk 3]
```diff
@@ -1,18 +1,22 @@
...
-#. 方位地图投影
-#. 圆柱地图投影
-#. 其它地图投影
+#. 极坐标投影（位于下图的 OTHER 分类）
+#. 地图投影：细分为四大类
+
+   - 圆柱地图投影（cylindrical projection）
+   - 圆锥地图投影（conic projection）
+   - 方位地图投影（azimuthal projection）
+   - 其它地图投影（miscellaneous projection，即下图的 thematic 分类）
```
Comment (core-man) [inline source/proj/index.rst:19]: GMT use the following order (see [GMT Map](https://docs.generic-mapping-tools.org/dev/cookbook/map-projections.html)):
> - 1. Conic projections
> - 2. Azimuthal projections
> - 3. Cylindrical projections
> - 4. Miscellaneous projections

File: source/proj/index.rst [hunk 4]
```diff
@@ -1,18 +1,22 @@
...
 ########
 
-GMT支持30余种不同的投影方式。这些投影方式可以分为三类：
+GMT 读取数据的实际坐标，并将其转换或者投影到图片的对应位置。GMT 支持 30 多种坐标变换或投影方式，
+这些投影方式可以分为三类，分类又细分为几个小类：
 
-#. 笛卡尔线性投影
-#. 极坐标投影
-#. 地图投影
+#. 笛卡尔投影：细分为三类（位于下图的 OTHER 分类）
```
Comment (core-man) [inline source/proj/index.rst:7]: Use **笛卡尔投影** intead of **笛卡尔线性投影**.

**Comment by core-man** [comment]
是否将投影汇总表格做成 include 文件，然后包含在此文件中？当前，该表格位于[-J](https://docs.gmt-china.org/latest/option/J/)中。

GMT 官方的汇总表格出现了两次：
- [Standardized command line options: -J option](https://docs.generic-mapping-tools.org/dev/cookbook/options.html#option-j)
- [MAP projection](https://docs.generic-mapping-tools.org/dev/cookbook/map-projections.html)

**Review by seisman — APPROVED** [review]

**Comment by seisman** [comment]
> 是否将投影汇总表格做成 include 文件，然后包含在此文件中？当前，该表格位于[-J](https://docs.gmt-china.org/latest/option/J/)中。
> 
> GMT 官方的汇总表格出现了两次：
> 
> * [Standardized command line options: -J option](https://docs.generic-mapping-tools.org/dev/cookbook/options.html#option-j)
> * [MAP projection](https://docs.generic-mapping-tools.org/dev/cookbook/map-projections.html):
> 
> 都是 include 了 [pro-codes.rst](https://github.com/GenericMappingTools/gmt/blob/master/doc/rst/source/proj-codes.rst_?rgh-link-date=2021-06-28T12%3A19%3A30Z):
> 
> ```
> .. include:: ../proj-codes.rst_
> ```

ok

File: source/proj/index.rst [hunk 5]
```diff
@@ -1,24 +1,30 @@
 地图投影
 ########
 
-GMT支持30余种不同的投影方式。这些投影方式可以分为三类：
+GMT 读取数据的实际坐标，并将其转换或者投影到图片的对应位置。GMT 支持 30 多种坐标变换或投影方式，
+这些投影方式可以分为三类，分类又细分为几个小类：
```
Comment (core-man) [inline source/proj/index.rst:5]:
```suggestion
GMT 读取数据的实际坐标，并将其转换或者投影到图片的对应位置。当前，GMT 支持 30 多种坐标变换或投影方式。
这些投影方式可以分为三大类，每一大类又可以细分为几个小类：
```

File: source/option/projection-codes.rst_ [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+   +----------------------------------------------------------+-------------------------------------------------------------+
+   | **-JW**\ [|lon0|/]\ *width*                              | Mollweide                                                   |
+   +----------------------------------------------------------+-------------------------------------------------------------+
+   | **-JX**\ *width*\ [**l**\|\ **p**\ *exp*\|\              |                                                             |
+   | **T**\|\ **t**][/\ *height*\ [**l**\|\ **p**\            |                                                             |
+   | *exp*\|\ **T**\|\ **t**]][**d**]                         | Linear, log\ :math:`_{10}`, :math:`x^a-y^b`, and time       |
+   +----------------------------------------------------------+-------------------------------------------------------------+
+   | **-JY**\ |lon0|/|lat0|\ *width*                          | Cylindrical equal area                                      |
+   +----------------------------------------------------------+-------------------------------------------------------------+
```
Comment (core-man) [inline source/option/projection-codes.rst_:71]:
```suggestion
   +----------------------------------------------------------+-------------------------------------------------------------+

```

File: source/proj/index.rst [hunk 7]
```diff
@@ -1,24 +1,30 @@
...
-#. 方位地图投影
-#. 圆柱地图投影
-#. 其它地图投影
+#. 极坐标投影（位于下图的 OTHER 分类）
+#. 地图投影：细分为四大类
+
+   - 圆柱地图投影（cylindrical projection）
+   - 圆锥地图投影（conic projection）
+   - 方位地图投影（azimuthal projection）
+   - 其它地图投影（miscellaneous projection，即下图的 thematic 分类）
```
Comment (core-man) [inline source/proj/index.rst:19]:
```suggestion
   - 其它地图投影（miscellaneous projection，即下图的 THEMATIC 分类）
```

File: source/proj/index.rst [hunk 8]
```diff
@@ -1,24 +1,30 @@
...
+
+   - 圆柱地图投影（cylindrical projection）
+   - 圆锥地图投影（conic projection）
+   - 方位地图投影（azimuthal projection）
+   - 其它地图投影（miscellaneous projection，即下图的 THEMATIC 分类）
 
 .. gmtplot:: projections.sh
    :width: 75%
    :show-code: false
    :caption: GMT支持的地图投影及坐标变换
```
Comment (core-man) [inline source/proj/index.rst:25]:
```suggestion

下表列出了 GMT 所支持的全部投影方式。

```

File: source/proj/index.rst [hunk 9]
```diff
@@ -1,24 +1,32 @@
 地图投影
 ########
 
-GMT支持30余种不同的投影方式。这些投影方式可以分为三类：
+GMT 读取数据的实际坐标，并将其转换或者投影到图片的对应位置。当前，GMT 支持 30 多种坐标变换或投影方式。
+这些投影方式可以分为三大类，每一大类又可以细分为几个小类：
```
Comment (core-man) [inline source/proj/index.rst:5]:
```suggestion
这些投影方式可以分为三类，每一类又可以细分为几个小类：
```

File: source/proj/index.rst [hunk 10]
```diff
@@ -1,24 +1,32 @@
...
+   - 笛卡尔线性投影
+   - 笛卡尔对数投影
+   - 笛卡尔指数投影
 
-#. 圆锥地图投影
-#. 方位地图投影
-#. 圆柱地图投影
-#. 其它地图投影
+#. 极坐标投影（位于下图的 OTHER 分类）
+#. 地图投影：细分为四大类
```
Comment (core-man) [inline source/proj/index.rst:14]:
```suggestion
#. 地图投影：细分为四类
```

**Pull request merged**